httpconnect/httpconnect.go

```diff
@@ -33,11 +34,34 @@ func IdleTimeoutSetter(i time.Duration) optSetter {
 	}
 }
 
+func AllowedPorts(ports []int) optSetter {
+	return func(f *HTTPConnectHandler) error {
+		f.allowedPorts = ports
+		return nil
+	}
+}
+
+func AllowedPortsFromCSV(csv string) optSetter {
+	return func(f *HTTPConnectHandler) error {
+		fields := strings.Split(csv, ",")
+		ports := make([]int, len(fields))
+		for i, f := range fields {
+			p, err := strconv.Atoi(f)
+			if err != nil {
+				return err
+			}
+			ports[i] = p
+		}
+		f.allowedPorts = ports
+		return nil
+	}
+}
+
 func New(next http.Handler, setters ...optSetter) (*HTTPConnectHandler, error) {
-	f := &HTTPConnectHandler{
-		errHandler: utils.DefaultHandler,
-		next:       next,
+	if next == nil {
```

Comment:
It makes more sense to fail soon than checking every time at `ServerHTTP()` and get unnoticed.